src/renderer/renderElement.ts

```diff
@@ -35,8 +35,79 @@ export interface ExcalidrawElementWithCanvas {
   canvasZoom: number;
   canvasOffsetX: number;
   canvasOffsetY: number;
+  nextCanvasZoom?: number; // for scheduleBatchedGenerateElementCanvas
 }
 
+const GENERATE_ELEMENT_CANVAS_BATCH_SIZE = 100;
+
+const lastRerenderScene = new WeakMap<CanvasRenderingContext2D, () => void>();
+
+export const scheduleBatchedGenerateElementCanvas = (
+  elements: readonly NonDeletedExcalidrawElement[],
+  context: CanvasRenderingContext2D,
+  sceneState: SceneState,
+  rerenderScene: () => void,
+) => {
+  lastRerenderScene.set(context, rerenderScene);
```

Comment:
what's the idea behind this?

Reply:
I tried to make the function not to depend on canvas 2d context singleton. It might be unnecessary complexity.